tests/model_registry/model_catalog/utils.py

```diff
@@ -1009,6 +1009,120 @@ def validate_items_sorted_correctly(items: list[dict], field: str, order: str) -
         raise ValueError(f"Invalid sort order: {order}")
 
 
+def verify_custom_properties_sorted(items: list[dict], property_field: str, sort_order: str) -> bool:
+    """
+    Verify if a list of items is sorted by a specific custom property value.
+
+    Expected sorting behavior:
+    1. Items WITH the custom property appear first, sorted by property value (ASC or DESC)
+    2. Items WITHOUT the custom property appear after, sorted by ID in ASC order
+
+    Args:
+        items: List of artifact items (with 'id' and 'customProperties')
+        property_field: Property field to sort by (e.g., "e2e_p90.double_value")
+        sort_order: "ASC" or "DESC" (applies only to items with the property)
+
+    Returns:
+        True if sorted correctly, False otherwise
+    """
+    if not items:
+        return True
+
+    property_name, value_type = property_field.rsplit(".", 1)
+    # Separate items into two groups
+    items_with_property, items_without_property = _split_items_by_custom_property(
+        items=items, property_name=property_name, value_type=value_type, property_field=property_field
+    )
+
+    LOGGER.info(
+        f"Total items: {len(items)}, with '{property_name}': {len(items_with_property)}, "
+        f"without: {len(items_without_property)}"
+    )
+
+    # Verify items with property come first
+    if items_with_property and items_without_property and items_with_property[-1][0] > items_without_property[0][0]:
```

Comment:
We should assert that items_with_property exists, otherwise we are just silently testing the id sorting works.

Reply:
this is a valid case, if I want to sort by A and A is not present we should fallback to ID sorting

Reply:
I understand. We need a way to tell this function when we are not expecting any match. That way we are not always testing with fallback to ID silently.